### ISSUE TITLE
Change the trusted indexers interface to expect a subgraph endpoint instead of an indexer-service endpoint

### ISSUE THREAD

**Issue by chriswessels** [edgeandnode/gateway#958]
Currently the trusted indexers endpoint expects indexer-service endpoints. Instead, trusted indexers should expect a list of network subgraph endpoints, which could be:
- gateway endpoints for access via decentralized network
- internal graph-node query node endpoints

This also has the benefit of enabling the operator to make use of the Subgraph ID in the endpoint URL, since this can be resolved by the Gateway, unlike indexer-service, which requires a hard coded Deployment ID.

**Comment by Theodus** [comment]
I agree that this would be a nice feature. The only major challenge is that the indexer-service API is different than the gateway & graph-node. So you won't be able to use the [`IndexerClient`](https://github.com/edgeandnode/gateway/blob/main/src/indexer_client.rs) the same way that we currently do in [src/network/subgraph_client.rs](https://github.com/edgeandnode/gateway/blob/main/src/network/subgraph_client.rs).
